src/Aspire.Hosting/ApplicationModel/ResourceNotificationService.cs

```diff
@@ -395,8 +395,17 @@ public Task PublishUpdateAsync(IResource resource, string resourceId, Func<Custo
     /// </summary>
     private static CustomResourceSnapshot UpdateHealthStatus(IResource resource, CustomResourceSnapshot previousState)
     {
-        // A resource is also healthy if it has no health check annotations and is in the running state.
-        if (previousState.HealthStatus is not HealthStatus.Healthy && !resource.TryGetAnnotationsIncludingAncestorsOfType<HealthCheckAnnotation>(out _) && previousState.State?.Text == KnownResourceStates.Running)
+        var hasHealthChecks = resource.TryGetAnnotationsIncludingAncestorsOfType<HealthCheckAnnotation>(out _);
+        var hasInitialSnapshot = resource.TryGetLastAnnotation<ResourceSnapshotAnnotation>(out _);
+
+        // A resource is also healthy if it
+        // - has no health check annotations
+        // - is in the running state
+        // - was not started with an initial snapshot
+        if (previousState.HealthStatus is not HealthStatus.Healthy
```

Comment:
If there is no initial snapshot, and there is no annotation, doesn't that mean `HealthStatus` must be null? That seems more specific test than not healthy.

Reply:
> HealthStatus must be null

Yeah, good point. There's nothing that could have affected the health status. Updated